x/ecocredit/server/core/send_test.go

```diff
@@ -0,0 +1,94 @@
+package core
+
+import (
+	"github.com/cosmos/cosmos-sdk/testutil/testdata"
+	"github.com/cosmos/cosmos-sdk/types"
+	"github.com/golang/mock/gomock"
+	"github.com/regen-network/regen-ledger/x/ecocredit"
+	"github.com/regen-network/regen-ledger/x/ecocredit/v1beta1"
+	"gotest.tools/v3/assert"
+	"testing"
+)
+
+func TestSend_Valid(t *testing.T) {
+	t.Parallel()
+	s := setupBase(t)
+	_, _, recipient := testdata.KeyTestPubAddr()
+	s.setupClassProjectBatch(t)
+	any := gomock.Any()
+	s.paramsKeeper.EXPECT().GetParamSet(any, any).Do(func(_ interface{}, p *ecocredit.Params) {
+		p.AllowlistEnabled = false
+		p.CreditClassFee = types.NewCoins(types.NewInt64Coin("foo", 20))
+		p.CreditTypes = []*ecocredit.CreditType{{Name: "carbon", Abbreviation: "C", Unit: "tonne", Precision: 6}}
+	}).AnyTimes()
```

Comment:
Not sure if it's necessary to check times here.
```suggestion
	}).Times(0)
```

Reply:
it gets called in sendEcocredits -> getCreditTypeFromBatchDenom